### PR TITLE
Add prow recording rules to build01

### PR DESCRIPTION
**Pull request by alvaroaleman**

### CODE REVIEW

File: clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml [hunk 1]
```diff
@@ -0,0 +1,147 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  annotations:
+    source: https://github.com/loodse/prow-dashboards/blob/e4af4c0ec879a0971591adcfc1e6f52c2ff015ee/prometheus-rules.yaml
+  name: prow-rules
+  namespace: openshift-monitoring
```
Comment (droslean) [inline clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml:7]: why this has to go to `openshift-monitoring` instead of `prow-monitoring`

Comment (alvaroaleman) [reply clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml:7]: There is no prow-monitoring on build01, it only exists on api.ci

Comment (stevekuznetsov) [reply clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml:7]: The data comes from the k8s prometheus, not ours, anyway

File: clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  - name: prow
+    rules:
+    # These metrics are based on generic pods, not just Prow jobs.
+
+    # group interesting information into a single metric
+    # squash metrics from multiple kube-state-metrics pods
+    # {namespace="...",pod="...",node="...",pod_ip="1.2.3.4",phase="..."} 1
+    - record: prow:pod
+      expr: |
+        max by (namespace, pod, node, pod_ip) (kube_pod_info) *
```
Comment (stevekuznetsov) [inline clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml:19]: Don't we a created-by-prow label selector here

File: clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+      expr: |
+        prow:pod:created_time *
+          on (namespace, pod)
+          group_left (org, repo, type, name, id)
+          prow:job:pending
+    # total memory used by a Prow job
+    # {namespace="...",pod="...",node="...",pod_ip="1.2.3.4",phase="...",org="...",repo="...",type="...",name="...",id="<UUID>"} [bytes]
+    - record: prow:job:memory_working_set_bytes
+      expr: |
+        sum by (namespace, pod, node, pod_ip) (container_memory_working_set_bytes{container="test"}) *
```
Comment (stevekuznetsov) [inline clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml:104]: In the past when we've been bitten by usage we got screwed on `sidecar` so why filter to only one container in the Pod?

Comment (stevekuznetsov) [reply clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml:104]: Also. max RSS is a separate metric and sensitive to spiky behavior. whereas this is not

**Comment by alvaroaleman** [comment]
> Why do you want these to be recording rules instead of a dashboard?

For performance reasons. Dashboard will just use the recording rules. Accompanying dashboards are here: https://github.com/loodse/prow-dashboards/tree/master/dashboards

File: clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml (hunk 2 above)
Comment (alvaroaleman) [reply clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml:19]: that requires another join with `kube_pod_labels`, do you know how to do that?

File: clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml (hunk 3 above)
Comment (alvaroaleman) [reply clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml]: added the container to the possible targets

File: clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml (hunk 2 above)
Comment (stevekuznetsov) [reply clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml:19]: You should just be able to do 

```
(kube_pod_info{label_created_by_prow="true"})
```

Comment (alvaroaleman) [reply clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml:19]: No, the label is not on the `kube_pod_info` metric, its only on `kube_pod_labels` and would require joining on namespace and name (And I don't know how joining in promQL works, the rules here are all copied)

Comment (stevekuznetsov) [reply clusters/build-clusters/01_cluster/assets/prom_prow_rules.yaml:19]: Ah, ew

**Pull request merged**